pkg/provider/aws/expose_test.go

```diff
@@ -389,3 +390,54 @@ func Test_CreateRestApi(t *testing.T) {
 
 	}
 }
+
+func Test_ConvertPath(t *testing.T) {
+	cases := []struct {
+		given          string
+		wantIfGreedy   string
+		wantIfNoGreedy string
+	}{
+		{
+			given:          `foo/bar`,
+			wantIfGreedy:   `foo/bar`,
+			wantIfNoGreedy: `foo/bar`,
+		},
+		{
+			given:          `foo/:bar`,
+			wantIfGreedy:   `foo/{bar}`,
+			wantIfNoGreedy: `foo/{bar}`,
+		},
+		{
+			given:          `foo/:bar*`,
+			wantIfGreedy:   `foo/{bar+}`,
+			wantIfNoGreedy: `foo/{bar}`,
+		},
+		{
+			given:          `foo/bar*`,
+			wantIfGreedy:   `foo/bar*`,
+			wantIfNoGreedy: `foo/bar*`,
+		},
+		{
+			given:          `foo//bar`,
+			wantIfGreedy:   `foo/bar`,
+			wantIfNoGreedy: `foo/bar`,
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.given, func(t *testing.T) {
+			for _, greedy := range []bool{true, false} { // https://www.youtube.com/watch?v=cwmDrQzsFT8#t=1m02s
+				t.Run(fmt.Sprintf(`greedy=%v`, greedy), func(t *testing.T) {
+					assert := assert.New(t)
+					actual := convertPath(tt.given, greedy)
+					var want string
+					if greedy {
+						want = tt.wantIfGreedy
+					} else {
+						want = tt.wantIfNoGreedy
+					}
+					assert.Equal(want, actual)
+				})
+			}
```

Comment:
very nitpick: I'd prefer just writing this out, doesn't seem like too much to warrant the for loop and not like we'd ever add more things to that list.
```go
assert := assert.New(t)

actualGreedy := convertPath(tt.given, true)
assert.Equal(tt.wantIfGreedy, actualGreedy, "greedy")

actualNotGreedy := convertPath(tt.given, false)
assert.Equal(tt.wantIfNoGreedy, actualNotGreedy, "not greedy")
```